ci2/ci/github.py

```diff
@@ -190,13 +190,45 @@ def __init__(self, branch):
 
         branch_gh_json = await gh.getitem(f'/repos/{repo_ss}/git/refs/heads/{self.branch.name}')
         new_sha = branch_gh_json['object']['sha']
+        if new_sha != self.sha:
+            self.sha = new_sha
+            if self.prs:
+                for pr in self.prs:
+                    pr.batch = None
+                    pr.passing = None
 
         new_prs = {}
         async for gh_json_pr in gh.getiter(f'/repos/{repo_ss}/pulls?state=open&base={self.branch.name}'):
             number = gh_json_pr['number']
-            pr = PR.from_gh_json(gh_json_pr)
-            await pr.refresh(gh)
+            if self.prs is not None and number in self.prs:
```

Comment:
I tried to be careful about using None to represent unknown.  In particular, I didn't want to start with {} and then have ci blow away all of the running jobs because it doesn't think there are any PRs.